packages/x-date-pickers/src/internals/hooks/useStaticPicker/useStaticPicker.tsx

```diff
@@ -46,14 +46,9 @@ export const useStaticPicker = <
         {...slotProps?.layout}
         slots={slots}
         slotProps={slotProps}
-        sx={[
-          ...(Array.isArray(sx) ? sx : [sx]),
-          ...(Array.isArray(slotProps?.layout?.sx)
-            ? slotProps!.layout!.sx
-            : [slotProps?.layout?.sx]),
-        ]}
-        className={clsx(className, slotProps?.layout?.className)}
-        ref={ref}
+        sx={mergeSx(providerProps.contextValue.rootSx, slotProps?.layout?.sx)}
```

Comment:
I'm using the context instead of direct prop access so that every component has the same data flow.
But these weird `providerProps.xxx` accesses will go away once we have the new structure.